### PR TITLE
Jc hold constant

### PR DESCRIPTION
**Pull request by jclagarde**
Made changes to hold T and pi constant on figure 2

### CODE REVIEW

**Comment by shak360** [comment]
image
This is what you get when you fix everything for figure 2, our simplest analysis. Uncensored, full populations of single lineages of increasing length.

**Comment by shak360** [comment]
image
Fiing everything for figure 3, uncensored populations of increasing size (populations with more lineages).

**Comment by shak360** [comment]
image

image

Figures 4 and 5 respectively. These figures are the same as above (2 -> 4) (3 -> 5) except that they are now censored by both time and fate.

File: lineage/BaumWelch.py [hunk 1]
```diff
@@ -101,14 +101,16 @@ def fit(tHMMobj, tolerance=np.spacing(1), max_iter=200):
...
                 cell_groups[state].append(lineage[ii])
-
-        # population wide pi calculation
-        tHMMobj.estimate.pi = pi_estimate / sum(pi_estimate)
-        # population wide T calculation
-        tHMMobj.estimate.T = T_estimate / T_estimate.sum(axis=1)[:, np.newaxis]
-        # opulation wide E calculation
-        for state_j in range(num_states):
-            tHMMobj.estimate.E[state_j] = tHMMobj.estimate.E[state_j].estimator([cell.obs for cell in cell_groups[state_j]])
+        if tHMMobj.estimate.fpi is None:
```
Comment (aarmey) [inline lineage/BaumWelch.py:104]: Why `None` and not true/false?

File: lineage/tHMM.py [hunk 2]
```diff
@@ -32,9 +42,12 @@ def __init__(self, X, num_states):
             FOM (str): For now, it is either "E": Exponential, or "G": Gompertz
             and it determines the type of distribution for lifetime of the cells
         """
+        fpi = kwargs.get('fpi', None)
```
Comment (aarmey) [inline lineage/tHMM.py:45]: You can just put `fT=None` in the arguments list to do the exact same thing.

File: lineage/tHMM.py [hunk 3]
```diff
@@ -6,19 +6,29 @@
 
 
 class estimate:
-    def __init__(self, num_states):
+    def __init__(self, num_states, **kwargs):
+        self.fpi = kwargs.get('fpi', None)
+        self.fT = kwargs.get('fT', None)
```
Comment (aarmey) [inline lineage/tHMM.py:11]: See my comment below.

File: lineage/Analyze.py [hunk 4]
```diff
@@ -79,19 +84,22 @@ def run_Analyze_over(list_of_populations, num_states, parallel=True):
     list_of_populations: a list of populations that contain lineages
     num_states: an integer number of states to identify (a hyper-parameter of our model)
     """
+    list_of_fpi = kwargs.get('list_of_fpi', [None]*len(list_of_populations))
```
Comment (aarmey) [inline lineage/Analyze.py:87]: This is the right way to handle this here, because the default is complex.

File: lineage/Analyze.py [hunk 5]
```diff
@@ -21,10 +21,12 @@ def preAnalyze(X, num_states):
     --------
     tHMMobj {obj}:
     """
-
+    fpi = kwargs.get('fpi', None)
```
Comment (aarmey) [inline lineage/Analyze.py:24]: Can just put this in the argument list with `fpi=None` and accomplish the same thing.

**Comment by shak360** [comment]
So, in general, is that method preferred over using the `**kwargs` in the function defintiion argument line? @aarmey

**Comment by aarmey** [comment]
@shak360 it is, if the defaults are going to be used within the function. If there are an unknown number of optional arguments, or most of the arguments are just being passed along to another function, then `**kwargs` is better.

**Pull request merged**